### ISSUE TITLE
Incorrect frame-to-sequence labelling, or image files, in dataset

### ISSUE THREAD

**Issue by bibbygoodwin** [facebookresearch/co3d#38]
Hi,

Unless I am mistaken, there are errors in the incorrect placement of image files within the dataset. I've only observed this on the Car category, but can't rule it out elsewhere. 

For instance, consider the car sequence '336_34852_64130'. This is associated with frame indexes, contiguously, from 9284 to 9385 (102 frames). However, an inspection of the image sequence reveals that about 20 of these frames come from a different sequence (see image for a subset - incorrect cars begin at 9306).
<img width="959" alt="Screenshot 2022-03-04 at 18 58 56" src="https://user-images.githubusercontent.com/28831431/156825931-33d19afa-1183-417e-9b7f-d29f09c925ef.png">

This issue does not seem to only apply to the RGB images. Depth maps and foreground probabilities (pictured here for frame 9307) are "correctly" linked to the RGB images, and thus incorrect for the sequence.
![download](https://user-images.githubusercontent.com/28831431/156825918-6155ad54-08b3-44c2-9213-790dcf860eae.png)

Would it be possible to look into this?
Thank you

**Comment by davnov134** [comment]
Hi, yes we are aware of this issue on the car category. This is because most car sequences also contain cars in the background which confuses the foreground mask detector. We are planning to release a better version of the annotations soon. In the meantime, you always have the access to the raw uncropped images (set the `box_crop` dataloader flag to `False`) so the segmentation masks are not an issue. The raw loaded depth maps should be correct, they would just be cropped wrongly so you can again use the uncropped depth.

**Comment by zzhuolun** [comment]
> Hi, yes we are aware of this issue on the car category. This is because most car sequences also contain cars in the background which confuses the foreground mask detector. We are planning to release a better version of the annotations soon. In the meantime, you always have the access to the raw uncropped images (set the `box_crop` dataloader flag to `False`) so the segmentation masks are not an issue. The raw loaded depth maps should be correct, they would just be cropped wrongly so you can again use the uncropped depth.

Thanks! I've also detected many wrong segmentation masks in the car sequences. E.g at [car/194_20898_41093](https://drive.google.com/file/d/1aJABWAFIx2nITmfH9hAJX-h0ztGTcD2o/view?usp=sharing)

 Could you confirm that this is mainly a issue for the car category? Or it also appears in other categories?

**Comment by davnov134** [comment]
So far we heard that cars are the main impacted category. A simple solution can be not to rely on the Pointrend segmentation masks and simply use the full image. Furthermore, the frame annotations containt Point_cloud_quality_score entries that are proportional to the quality of the reconstructed point cloud. For this sequence, I am fairly sure that the score is going to be low.

**Comment by zzhuolun** [comment]
> So far we heard that cars are the main impacted category. A simple solution can be not to rely on the Pointrend segmentation masks and simply use the full image. Furthermore, the frame annotations containt Point_cloud_quality_score entries that are proportional to the quality of the reconstructed point cloud. For this sequence, I am fairly sure that the score is going to be low.

Thanks for the reply. The averge point cloud score of car is indeed low (-0.68), though still higher than that of some other categories, as show bellow. It could be that though some of the segmentation masks are wrong, but the final pointcloud is good, e.g at the sequence car/206_21810_45890.
![output](https://user-images.githubusercontent.com/23330740/158608139-2fcc12a5-6305-40fb-a52b-b71420366062.png)

In the CO3D paper it is mentioned that of all the 22665 collected videos, only 5625 of them have accurate point clouds. What threshold of the point cloud score do you use for determing if a sequence is accurate or not?

**Comment by davnov134** [comment]
The test set is only composed of sequences that passed the test. We did not remove the sequences with inaccurate point clouds, only those with inaccurate cameras. It is not reasonable to remove all point-cloud inaccurate sequences especially if all methods in the paper do not require point clouds for training (all methods are supervised with segmentation mask error and the photometric error only). All point clouds with the accuracy score >0.0 were considered accurate.
